httpx/dispatch/http2.py

```diff
@@ -116,16 +88,113 @@ def init_complete(self) -> BaseEvent:
         # Some websites (*cough* Yahoo *cough*) balk at this setting being
         # present in the initial handshake since it's not defined in the original
         # RFC despite the RFC mandating ignoring settings you don't know about.
-        del self.h2_state.local_settings[
-            h2.settings.SettingCodes.ENABLE_CONNECT_PROTOCOL
-        ]
+        del self.state.local_settings[h2.settings.SettingCodes.ENABLE_CONNECT_PROTOCOL]
+
+        self.state.initiate_connection()
+        data_to_send = self.state.data_to_send()
+        await self.socket.write(data_to_send, timeout)
+
+    @property
+    def is_closed(self) -> bool:
+        return False
+
+    def is_connection_dropped(self) -> bool:
+        return self.socket.is_connection_dropped()
+
+    async def close(self) -> None:
+        await self.socket.close()
+
+    async def wait_for_outgoing_flow(self, stream_id: int, timeout: Timeout) -> int:
+        """
+        Returns the maximum allowable outgoing flow for a given stream.
+
+        If the allowable flow is zero, then waits on the network until
+        WindowUpdated frames have increased the flow rate.
+
+        https://tools.ietf.org/html/rfc7540#section-6.9
+        """
+        local_flow = self.state.local_flow_control_window(stream_id)
+        connection_flow = self.state.max_outbound_frame_size
+        flow = min(local_flow, connection_flow)
+        while flow == 0:
+            await self.receive_events(timeout)
+            local_flow = self.state.local_flow_control_window(stream_id)
+            connection_flow = self.state.max_outbound_frame_size
+            flow = min(local_flow, connection_flow)
+        return flow
```

Comment:
Some duplicated lines here - possible refactor?

```python
        while True:
            local_flow = self.state.local_flow_control_window(stream_id)
            connection_flow = self.state.max_outbound_frame_size
            flow = min(local_flow, connection_flow)
            if flow == 0:
                await self.receive_events(timeout)
                continue
            return flow
```

I can see that the current implementation really fits the docstring though, so not particularly urging about this.

Reply:
Yup, may well be worthwhile. I’ll get this in as-is for now, and can follow up.